src/providers/EtherspotContextProvider.tsx

```diff
@@ -364,6 +364,41 @@ const EtherspotContextProvider = ({
     [sdk, accountAddress]
   );
 
+  const getRatesByTokenAddresses = useCallback(
+    async (chainId: number, tokenAddresses: string[]) => {
+      let tokens = tokenAddresses.filter((address) => address !== null);
+      if (!sdk || !tokens.length) return null;
+      try {
+        const rates: RateData = await sdk.fetchExchangeRates({ tokens, chainId });
+        if (!rates.errored && rates.items.length) {
+          let rateByAddress: Record<string, number> = {};
+          rates.items.forEach((rate) => {
+            rateByAddress[rate.address] = rate.usd;
+          });
+          return rateByAddress;
+        }
```

Comment:
can be just

if (rates.errored || !rates?.items?.length) return null;

then no extra if wrap